src/main/java/edu/cornell/kfs/fp/businessobject/USBankRecordFieldUtils.java

```diff
@@ -0,0 +1,194 @@
+ */
+package edu.cornell.kfs.fp.businessobject;
+
+import java.io.IOException;
+import java.sql.Date;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+/**
+ * Provides methods for parsing USBank addendum records.
+ * In the long-run, it may be wise/useful to turn this into a subclass of 
+ * org.kuali.kfs.sys.businessobject.BusinessObjectStringParserFieldUtils and 
+ * provide subclasses that assist in parsing, but we won't for now.
+ *
+ */
+public class USBankRecordFieldUtils {
+
+  /**
+   * Extracts a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @return The trimmed substring
+   * @throws StringIndexOutOfBoundsException May occur when taking the actual substring if the provided bounds are beyond the limits of the superstring
+   */
+  public static String extractNormalizedString(String line, int begin, int end) throws StringIndexOutOfBoundsException {
+    String theString = line.substring(begin, end);
+    if(theString.trim().length()==0) {
+      return null;
+    }
+    return theString;
+  }
+
+  /**
+   * Extracts a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @param required True if the value is required to not be empty
+   * @param lineCount The current line number
+   * @return The trimmed substring
+   * @throws IOException When a required value is missing
+   * @throws StringIndexOutOfBoundsException When taking the actual substring if the provided bounds are beyond the limits of the superstring
+   */
+  public static String extractNormalizedString(String line, int begin, int end, boolean required, int lineCount) throws IOException {
+    String theValue = extractNormalizedString(line, begin, end);
+    if (required) {
+      if (theValue==null) {
+        throw new IOException("A required value was missing at " + begin + " " + end + " on line " + lineCount);
+      }
+    }
+    return theValue;
+  }
+
+  /**
+   * Extracts a Date from a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @param lineCount The current line number
+   * @return The Date parsed from the trimmed substring
+   * @throws Exception When unable to parse the date
+   */
+  public static Date extractDate(String line, int begin, int end, int lineCount) throws Exception {
+    Date theDate;
+    try {
+        String sub = line.substring(begin, end);
+        String year = sub.substring(0,4);
+        String month = sub.substring(4,6);
+        String day = sub.substring(6,8);
+        theDate = Date.valueOf(year+"-"+month+"-"+day);
+    }
+    catch (Exception e) {
```

Comment:
I was keeping it the same as the method it replaced, but we can probably throw it as an IOException. What do you think?